src/Package/Impl/ProjectSystem/Commands/SourceFilesCommand.cs

```diff
@@ -73,7 +73,7 @@ internal sealed class SourceFilesCommand : SendFileCommandBase, IAsyncCommandGro
                         string projectName = properties.GetProjectName();
                         string remotePath = await properties.GetRemoteProjectPathAsync();
 
-                        await SendToRemoteAsync(files, projectDir, projectName, remotePath, CancellationToken.None);
+                        await SendToRemoteAsync(files, projectDir, projectName, remotePath);
```

Comment:
Operation cancel works well, while we are in transfer and extract phase. While in compression phase, if we are already compressing a massive file, it does not cancel until processing on the file is done. 
VS UI is blocked from accepting any input while transfer is in progress.